src/app/components/dialogUpdateTranslation/DialogUpdate.tsx

```diff
@@ -0,0 +1,124 @@
+import { UpdateTranslation } from "@/app/Service/DictionariesServices";
+import { schemaUpdateTranslation } from "@/app/Utils/UpdateValidation";
+import { yupResolver } from "@hookform/resolvers/yup";
+import {
+  Button,
+  Chip,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  Select,
+  TextField,
+  Box,
+  DialogActions,
+  MenuItem,
+} from "@mui/material";
+import { useForm } from "react-hook-form";
+import { useQueryClient } from "react-query";
+import { toast } from "react-toastify";
+
+interface UpdateDataForm {
+  key: string;
+  text: string;
+  UpdateLanguages: string[];
+}
+
+const DialogUpdate = ({
+  open,
+  handleClose,
+  IdProject,
+  translationData,
+  languages,
+}: {
+  open: boolean;
+  handleClose: () => void;
+  IdProject: any;
+  translationData: any;
+  languages: any;
+}) => {
+  const queryClient = useQueryClient();
+
+  const languageKeys = Object.keys(languages);
+  // console.log(languageKeys);
+  
+
+  const { register: update, handleSubmit: handleSubmitUpdate } =
+    useForm<UpdateDataForm>({
+      resolver: yupResolver(schemaUpdateTranslation),
+      defaultValues: {
+        key: "",
+        text: "",
+        UpdateLanguages: [],
+      },
+    });
+
+  const SubmitUpdate = async (
```

Comment:
you use IdProject and IdTranslation as params but they exist into the component you should remove them from the params like that : 
```
const SubmitUpdate = async (
    dataUpdate: UpdateDataForm
  ) => {
    try {
      const data = await UpdateTranslation(
        IdProject,
        translationData.id,
        dataUpdate
      );
      // queryClient.invalidateQueries("allTranslation");
      console.log(data);
      return data;
    } catch (error: any) {
      toast.error(error.response?.data?.message || "An error occurred");
    }
  };
```